### PR TITLE
Move mission to robot interface

### LINKED ISSUE #415: Add 'initiate_mission' function in RobotInterface
**Describe the new feature you would like to see**
An optional path in the state machine will be to start a full mission on a robot, taking advantage of the supplier's mission system.
The RobotInterface has to support this, so we need to add a function for `intitiate_mission`.
This should raise a `NotImplementedError` by default, like the `initiate_step` function does.

**How will this feature affect the current Threat Model?**
N/A

### PR DESCRIPTION
**Pull request by Christdej**
Closes #415

### CODE REVIEW

**Comment by Christdej** [comment]
Requires a new version and changes in robot-repos to avoid breaking

**Review by tsundvoll — CHANGES_REQUESTED** [review]
Overall the changes looks good, just a few comments.

File: src/isar/models/communication/message.py [hunk 1]
```diff
@@ -2,11 +2,13 @@
 from typing import Optional
 
 from alitra import Pose
+from isar.models.mission_metadata.mission_metadata import MissionMetadata
```
[the remark this answers is not in the document]
Comment (tsundvoll) [reply src/isar/models/communication/message.py:5]: I agree, but this is how isort sort on my side. Might be some PyCharm isort setting that are strange, or that the scope of the editor is set to one level above isar.

[the remark this answers is not in the document]
Comment (aestene) [reply src/isar/models/communication/message.py:5]: Alternatively, there may be some isort settings on your side which hasn't been properly adjusted to the correct one that you will find in PyCharm... Tricky question this :D

[the remark this answers is not in the document]
Comment (aestene) [reply src/isar/models/communication/message.py:5]: I resaved the file and it grouped them together so I'm not sure what happened.

File: src/isar/services/utilities/scheduling_utilities.py [hunk 2]
```diff
@@ -135,7 +136,12 @@ def verify_state_machine_ready_to_receive_mission(self, state: States) -> bool:
 
         return is_state_machine_ready_to_receive_mission
 
-    def start_mission(self, mission: Mission, initial_pose: Optional[Pose]) -> None:
+    def start_mission(
+        self,
+        mission: Mission,
+        mission_metadata: MissionMetadata,  # type: ignore
```
[the remark this answers is not in the document]
Comment (aestene) [reply src/isar/services/utilities/scheduling_utilities.py:142]: Removed

File: tests/isar/services/readers/test_mission_reader.py [hunk 3]
```diff
@@ -63,21 +69,19 @@ def test_read_mission_from_file(mission_reader) -> None:
...
-        expected_mission.metadata.vertical_reference_system
-        == mission.metadata.vertical_reference_system
+        expected_metadata.coordinate_reference_system
+        == MissionMetadata.coordinate_reference_system
     )
     assert (
-        expected_mission.metadata.data_classification
-        == mission.metadata.data_classification
+        expected_metadata.vertical_reference_system
+        == MissionMetadata.vertical_reference_system
```
[the remark this answers is not in the document]
Comment (aestene) [reply tests/isar/services/readers/test_mission_reader.py:82]: Removed them

File: tests/isar/state_machine/test_state_machine.py [hunk 4]
```diff
@@ -90,9 +92,11 @@ def test_reset_state_machine(state_machine) -> None:
 def test_state_machine_transitions(injector, state_machine_thread) -> None:
     step: Step = DriveToPose(pose=MockPose.default_pose)
     mission: Mission = Mission(tasks=[Task(steps=[step])])  # type: ignore
-
+    metadata: MissionMetadata = MissionMetadata(mission.id)
```
[the remark this answers is not in the document]
Comment (aestene) [reply tests/isar/state_machine/test_state_machine.py:95]: Added new issue in #438

File: src/isar/apis/schedule/scheduling_controller.py [hunk 5]
```diff
@@ -228,3 +238,18 @@ def get_info(self):
...
+            tasks=[self._task_api_response(task) for task in mission.tasks],
+        )
+
+    def _task_api_response(self, task: Task) -> TaskResponse:
+        return TaskResponse(
+            id=task.id,
+            tag_id=task.tag_id,
+            steps=list(
+                map(lambda x: {"id": x.id, "type": x.__class__.__name__}, task.steps)
+            ),
```
[the remark this answers is not in the document]
Comment (aestene) [reply src/isar/apis/schedule/scheduling_controller.py:254]: I wrote out the for-loop instead.

**Pull request merged**